### DISCUSSION TITLE
Linux Mint 22 and Regolith 3.2 beta not playing well together

### DISCUSSION THREAD

**Discussion by GeoSpark** [regolith-linux/regolith-desktop#1036] [Q&A] [answered]
I have recently upgraded my Linux Mint 21.3 to 22, which is based on Ubuntu 24.04. I had been using Regolith2 for a while and so decided to upgrade to version 3 at the same time. I have Xfce installed as the original desktop running on X11, on a dual monitor setup and an Nvidia RTX 3050 GPU running proprietary drivers.

I did a complete clean of Regolith2 before installing Regolith3 as per the instructions. After a reboot, I get the option for "Regolith/X11" but when I log in, I get a blank screen for a few seconds and then thrown back to the login screen. I've read around and found that others have had similar issues, but I can't find the magic invocation I need to fix the problem.

I tried cleaning out the local config directory, and a new one doesn't get created in its place, and I can see nothing in syslog that might indicate the source of the problem, but I could easily be missing something.

Where would be a good place to start looking for the cause of the issue?

Edit:
I've done a full fresh reinstall of Mint 22 Cinnamon and Regolith 3.2, but I get the same problem. I installed the sway version too, and although I get the option to use Regolith/Wayland, it does the same thing.
Running Cinnamon with either X11 or Wayland works fine.
Choosing the i3 login option also works fine, so it seems to be a Regolith configuration problem, rather than a fundamental i3 problem.

**Comment by GeoSpark** [comment, marked as answer]
So the issue is an unrecognised command line switch in `/usr/bin/regolith-session`, on the lines that call `gnome-session` the switch is `--builtin`. This seems to be because it's an old version of `gnome-session` that [doesn't support that switch](https://github.com/i3-gnome/i3-gnome/issues/50).

However, the root cause is that the test on line 28 of `regolith-session` tests for `"$VERSION_CODENAME" == "noble"`, but on Mint at least, `$VERSION_CODENAME` is set to `wilma` and `$UBUNTU_CODENAME` is set to `noble`.  So it was falling through to the `else` block which uses the `--builtin` flag.

I have fixed it locally by changing the test to use `$UBUNTU_CODENAME`, and it now works fine.

**Reply by kgilmer** [reply]
Hi @GeoSpark , can you provide a diff or your version of `/usr/bin/regolith-sesssion`?  I may be able to integrate your fix before 3.2 release is complete..

**Reply by GeoSpark** [reply]
Sure:
```
--- ./regolith-session	2024-07-31 13:05:38.826605069 +0100
+++ /usr/bin/regolith-session	2024-07-31 13:11:43.132554700 +0100
@@ -25,7 +25,7 @@
 
     if [[ "$ID" == "arch" ]]; then 
         exec gnome-session --session=$SESSION_TYPE --disable-acceleration-check "$@"
-    elif [[ "$VERSION_CODENAME" == "bionic" || "$VERSION_CODENAME" == "buster" || "$VERSION_CODENAME" == "noble" || "$VERSION_CODENAME" == "trixie" ]]; then
+    elif [[ "$VERSION_CODENAME" == "bionic" || "$VERSION_CODENAME" == "buster" || "$UBUNTU_CODENAME" == "noble" || "$VERSION_CODENAME" == "noble" || "$VERSION_CODENAME" == "trixie" ]]; then
         exec gnome-session --session=$SESSION_TYPE --disable-acceleration-check "$@"
     else
         exec gnome-session --builtin --session=$SESSION_TYPE --disable-acceleration-check "$@"
```

**Reply by kgilmer** [reply]
I opted to switch thing a bit so by default the new behavior is run.  LMK if you see any concerns here @GeoSpark 

https://github.com/regolith-linux/regolith-session/pull/40
